spec/datadog/di/transport_spec.rb

```diff
@@ -0,0 +1,96 @@
+require "datadog/di/spec_helper"
+require "datadog/di/transport"
+
+RSpec.describe Datadog::DI::Transport do
+  let(:agent_settings) do
+    double('agent settings')
+  end
+
+  describe '.new' do
+    it 'creates an instance using agent settings' do
+      expect(agent_settings).to receive(:hostname).and_return('localhost')
+      expect(agent_settings).to receive(:port).and_return(8126)
+      expect(agent_settings).to receive(:timeout_seconds).and_return(1)
+      expect(agent_settings).to receive(:ssl).and_return(false)
+
+      expect(described_class.new(agent_settings)).to be_a(described_class)
+    end
+  end
+
+  # These are fairly basic tests. The agent will accept all kinds of
+  # semantically nonsensical payloads. The tests here are useful to
+  # ascertain that things like content type is set correctly for each
+  # endpoint.
+  #
+  # Realistically, the only test that can check that the payload being
+  # sent is the correct one is a system test.
+  describe 'send methods' do
+    skip_unless_integration_testing_enabled
+
+    let(:port) { 8126 }
+
+    before do
+      expect(agent_settings).to receive(:hostname).and_return('localhost')
+      expect(agent_settings).to receive(:port).and_return(port)
+      expect(agent_settings).to receive(:timeout_seconds).and_return(1)
+      expect(agent_settings).to receive(:ssl).and_return(false)
+    end
+
+    let(:client) do
+      described_class.new(agent_settings)
+    end
+
+    describe '.send_diagnostics' do
+      let(:payload) do
+        {}
+      end
+
+      it 'does not raise exceptions' do
+        expect do
+          client.send_diagnostics(payload)
+        end.not_to raise_exception
+      end
```

Comment:
There is not much value in the `let` statement, so we can shrink entire test to this:

```suggestion
      it 'does not raise exceptions' do
        expect { client.send_diagnostics({}) }.not_to raise_exception
      end
```

Reply:
I want it to be clear what the payload is, hence it is declared separately.